src/main/java/com/limechain/network/ConnectionManager.java

```diff
@@ -122,18 +164,31 @@ public void updatePeer(PeerId peerId, BlockAnnounceMessage blockAnnounceMessage)
         }
     }
 
-    public void removePeer(PeerId peerId) {
-        peers.remove(peerId);
-    }
-
+    /**
+     * Check if we have an open GRANDPA responder stream with peer.
+     *
+     * @param peerId peer to check
+     * @return do peer info and GRANDPA responder stream exist
+     */
     public boolean isGrandpaConnected(PeerId peerId) {
-        return peers.containsKey(peerId) && peers.get(peerId).getGrandpaStreams() != null;
+        return peers.containsKey(peerId) && peers.get(peerId).getGrandpaStreams().getResponder() != null;
     }
 
+    /**
+     * Check if we have an open Block Announce responder stream with peer.
+     *
+     * @param peerId peer to check
+     * @return do peer info and Block Announce responder stream exist
+     */
     public boolean isBlockAnnounceConnected(PeerId peerId) {
-        return peers.containsKey(peerId) && peers.get(peerId).getBlockAnnounceStreams() != null;
+        return peers.containsKey(peerId) && peers.get(peerId).getBlockAnnounceStreams().getResponder() != null;
     }
 
+    /**
+     * Get the ids of all peers with open connections.
+     *
+     * @return set of connected peer ids
+     */
```

Comment:
We should probably be explicit about whether open connections mean either block announce or grandpa
```suggestion
    /**
     * Gets the ids of all peers with open connections. Open connection means either Grandpa or Block Announce stream has been opened.
     *
     * @return set of connected peer ids
     */
```